README.tfdoc.hcl

```
@@ -0,0 +1,531 @@
+          readme_type = "list(access)"
+          default     = []
+          description = <<-END
+            An array of objects that define dataset access for one or more entities.
+          END
+
+          attribute "domain" {
+            type        = string
+            description = <<-END
+              A domain to grant access to. Any users signed in with the domain specified will be granted the specified access.
+            END
+          }
+
+          attribute "role" {
+            type        = string
+            description = <<-END
+              Describes the rights granted to the user specified by the other member of the access object. Basic, predefined, and custom roles are supported. Predefined roles that have equivalent basic roles are swapped by the API to their basic counterparts.
+            END
+          }
+
+          attribute "group_by_email" {
+            type        = string
+            description = <<-END
+              An email address of a Google Group to grant access to.
+            END
+          }
+
+          attribute "user_by_email" {
+            type        = string
+            description = <<-END
+              An email address of a Google User to grant access to.
+            END
+          }
+
+          attribute "special_group" {
+            type        = string
+            description = <<-END
+              A special group to grant access to. Possible values include:
+              - `projectOwners`: Owners of the enclosing project.
+              - `projectReaders`: Readers of the enclosing project.
+              - `projectWriters`: Writers of the enclosing project.
+              - `allAuthenticatedUsers`: All authenticated BigQuery users.
+            END
+          }
+        }
+
+        variable "view" {
+          type        = any
+          readme_type = "object(view)"
+          default     = []
+          description = <<-END
+            A view from a different dataset to grant access to.
+          END
+
+          attribute "project_id" {
+            required    = true
+            type        = string
+            description = <<-END
+              The ID of the project containing this table.
+            END
+          }
+
+          attribute "table_id" {
+            required    = true
+            type        = string
+            description = <<-END
+              The ID of the dataset containing this table.
+            END
+          }
+
+          attribute "dataset_id" {
+            required    = true
+            type        = string
+            description = <<-END
+              The ID of the table.
+            END
+          }
+        }
+
+        variable "role" {
+          type = any
+          readme_type = <<-END
+            list(object({
+            role           = string
+            special_group  = string
+            group_by_email = string
+            user_by_email  = string
+          }))
+          END
```

complex types are just `list(type)`  we do not expand objects as we explain eahc attribute anyway.

is role a list of roles? looks like the naming is bad but can be fixed at another time..



```suggestion
          readme_type = <<-END
            list(object({
            role           = string
            special_group  = string
            group_by_email = string
            user_by_email  = string
          }))
          END
```